plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsSchema.php

```diff
@@ -143,6 +153,189 @@ public function validate_schema( $options ) {
 			return false;
 		}
 
+		$other_schema = [
+			'required',
+			'hidden',
+		];
+
+		foreach ( $other_schema as $rule ) {
+			if ( empty( $options['rules'][ $rule ] ) ) {
+				continue;
+			}
+			if ( ! is_array( $options['rules'][ $rule ] ) ) {
+				$property_error = sprintf( 'The %s rules must be an array.', $rule );
+				$message        = sprintf( 'Unable to register field with id: "%s". %s', $id, $property_error );
+				_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), esc_html( $this->release_version ) );
+				return false;
+			}
+			$result = $validator->validate(
+				Helper::toJSON(
+					[
+						'$schema'    => 'http://json-schema.org/draft-07/schema#',
+						'type'       => 'object',
+						'properties' => [
+							'test' => $options['rules'][ $rule ],
+						],
+						'required'   => [ 'test' ],
+					]
+				),
+				$this->meta_schema_json
+			);
+			if ( $result->hasError() ) {
+				$message = sprintf( 'Unable to register field with id: "%s". %s', $options['id'], $result->error() );
+				_doing_it_wrong( 'woocommerce_register_additional_checkout_field', esc_html( $message ), esc_html( $this->release_version ) );
+				return false;
+			}
+		}
```

Comment:
Is this botched rebase? It looks the the validation ran beforehand. Or is this somehow needed too?

Reply:
possibly botched